fenix/app/src/main/java/org/mozilla/fenix/components/metrics/FontParser.kt

```diff
@@ -0,0 +1,251 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.DataInputStream
+import java.io.FileInputStream
+import java.io.IOException
+import java.io.InputStream
+import java.security.MessageDigest
+import java.util.Objects
+import kotlin.math.min
+
+/**
+ * Parse a font, given via an InputStream, to extract the Font information
+ * including Family, SubFamily, Revision, etc
+ */
+@Suppress("MagicNumber")
+class FontParser(inputStream: InputStream) {
```

Comment:
Consideration: It seems like this constructor is only used for testing, so let's replace the secondary one with this, and create a method to open and set the stream that is internal and visible for testing only?

Reply:
In Kotlin, you don't need to do these java-style bindings, you can make properties through constructors.

```suggestion
class FontParser(private val inputStream: InputStream) {
```

Based on the comment above, you can re-work this to be more concise.